js/webapp.js

```diff
@@ -182,15 +182,11 @@ const staticHandlers = {
  *  This is passed to every other service of the plugin, so that 
  *  the service can be called by other services under the plugin
  */
-function WebServiceHandle(urlPrefix, httpPort, httpsPort) {
+function WebServiceHandle(urlPrefix, networkConfig) {
   this.urlPrefix = urlPrefix;
-  if (httpsPort) {
-    this.port = httpsPort;
-    this.isHttps = true;
-  } else {
-    this.port = httpPort;
-    this.isHttps = false;
-  }
+  this.isHttps = networkConfig.isHttps;
+  this.port = networkConfig.port;
+  this.host = networkConfig.host;
```

Comment:
Having a hostname in every service handle implies that different handles may be for different hosts, so it suggests a distributed architecture (which we don't have and don't plan). I'd rather have this as a global variable or constant.

Reply:
OK. The comments also note that this whole thing should go away if we can route directly some day.

Reply:
While we are making network calls to ourselves, we have to make sure that we make a call to one of the addresses on which we are listening, which may not include any loopback addresses. I haven't read the code, but this discussion makes me ask the question: do we account for that?

Reply:
Yes. If no loopback is found, the first address is used instead.